### PR TITLE
Replace default JSON engine (Poison) with Jason

### PR DESCRIPTION
**Pull request by indrekj**
Almost all libraries nowadays use Jason. It's still possible to use Poison
but it is no longer default.

### CODE REVIEW

File: lib/logstash_logger_formatter.ex [hunk 1]
```diff
@@ -114,4 +114,12 @@ defmodule LogstashLoggerFormatter do
   defp add_message(event, message) do
     Map.put(event, @msg_field, to_string(message))
   end
+
+  defp encode_to_iodata!(payload) do
+    if function_exported?(@engine, :encode_to_iodata!, 1) do
```
Comment (take-five) [inline lib/logstash_logger_formatter.ex:119]: This check will be performed in runtime every time something is logged. But `@engine` is a compile-time property. What do you think of running this check at compile-time too?

Comment (indrekj) [reply lib/logstash_logger_formatter.ex]: https://github.com/salemove/logstash_logger_formatter/compare/11146dabac146818514159be6980091f5da0d87e..ef0c103147a248e7dc29a83b65833a198393e49c

**Review by take-five — APPROVED** [review]

**Review by urmastalimaa — APPROVED** [review]

**Comment by indrekj** [comment]
@take-five can you publish the new version, I don't have permissions?

**Comment by take-five** [comment]
I think you should have access to Hex.pm Okta app. Can you add yourself as a publisher?

**Comment by indrekj** [comment]
I got in, but the webpage does not allow adding a new owner. Has to be done on the command line if I understood correctly.

**Comment by take-five** [comment]
Published

**Comment by indrekj** [comment]
thx

**Pull request merged**